packages/driver/src/cypress/error_messages.ts

```diff
@@ -1393,16 +1393,20 @@ export default {
       message: `${cmd('select')} was called with an array of arguments but does not have a \`multiple\` attribute set.`,
       docsUrl: 'https://on.cypress.io/select',
     },
+    invalid_number: {
+      message: `${cmd('select')} was called with an invalid index: \`{{index}}\`. Index must be a non-negative integer.`,
+      docsUrl: 'https://on.cypress.io/select',
+    },
     multiple_elements: {
       message: `${cmd('select')} can only be called on a single \`<select>\`. Your subject contained {{num}} elements.`,
       docsUrl: 'https://on.cypress.io/select',
     },
     multiple_matches: {
-      message: `${cmd('select')} matched more than one \`option\` by value or text: \`{{value}}\``,
+      message: `${cmd('select')} matched more than one \`option\` by value, index, or text: \`{{value}}\``,
       docsUrl: 'https://on.cypress.io/select',
     },
     no_matches: {
-      message: `${cmd('select')} failed because it could not find a single \`<option>\` with value or text matching: \`{{value}}\``,
+      message: `${cmd('select')} failed because it could not find a single \`<option>\` with value, index, or text matching: \`{{value}}\``,
```

Comment:
Can you add a test for the index case here? Trying to do `cy.select(3)` when there are only 2 options or w/e.